docs/auth.md

```diff
@@ -175,6 +175,32 @@ generated containing the credentials configured in the `Secret`, and these
 credentials are then used to authenticate when retrieving any
 `PipelineResources`.
 
+## GitHub Apps
+
+[GitHub Apps](https://developer.github.com/apps/about-apps/) are the preferred way for automated operations say from
+within a pipeline to interact with GitHub, rather than managing a 'bot' user and personal access token.
+
+A GitHub App is installed into a GitHub Organisation and an API is used to retrieve a token that can then be used to
+authenticate Git requests.  This means we now have many tokens, one for each Organisation.  For Tekton to mount the correct
+Kubernetes secret for a given Organisation we need to annotate each secret with the owner that matches the Organisation,
+so we can filter out other secrets during the `creds-init` pipeline step.
+
+   ```yaml
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: basic-user-pass
+     annotations:
+       tekton.dev/git-0: https://github.com # Described below
+       tekton.dev/githubapp-owner: <github organisation>
```

Comment:
We should also include App ID and Installation ID, this way one cluster could potentially reference multiple Apps.

Reply:
Ok cool, I've not looked at using multiple GitHub Apps for a single Tekton deployment yet.  I wonder if we could do that as a follow up PR so that we keep this one a bit simpler?